backend/courses/util.py

```diff
@@ -506,6 +512,73 @@ def upsert_course_from_opendata(info, semester, missing_sections=None):
         missing_sections.discard(section.full_code)
 
 
+def add_attributes(section, attributes):
+    """
+    Add attributes to course of section.
+    Create attribute if it does not exist
+    """
+    for attribute in attributes:
+        try:
```

Comment:
This could be refactored to:
```
attr = Attribute.objects.get_or_create(
    code=attribute.get("attribute_code"),
    defaults={
        "description": attribute.get("attribute_desc"),
        "school": school
    }
)
```